internal/test/framework/windowsvm.go

```diff
@@ -146,6 +149,68 @@ func (w *windowsVM) CopyFile(filePath, remoteDir string) error {
 	return nil
 }
 
+// RetrieveFiles retrieves list of files from remote directory to the local directory.
+// The implementation can be changed if the use-case arises.
+func (w *windowsVM) RetrieveFiles(remoteDir, localDir string) error {
+	if w.sshClient == nil {
+		return fmt.Errorf("RetrieveFile cannot be called without a ssh client")
+	}
+
+	// Create local dir
+	err := os.MkdirAll(localDir, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("could not create %s: %s", localDir, err)
+	}
+
+	sftp, err := sftp.NewClient(w.sshClient)
+	if err != nil {
+		return fmt.Errorf("sftp initialization failed: %v", err)
+	}
+	defer sftp.Close()
+
+	// Get the list of all files in the directory
+	remoteFiles, err := sftp.ReadDir(remoteDir)
+	if err != nil {
+		return fmt.Errorf("error opening remote file: %v", err)
+	}
+
+	for _, remoteFile := range remoteFiles {
+		// Assumption: We ignore the directories here the reason being RetrieveFiles should just retrieve files
+		// in a directory, if this is directory, we should have called RetrieveFiles on this directory
+		if remoteFile.IsDir() {
+			continue
+		}
+		fileName := remoteFile.Name()
+		dstFile, err := os.Create(filepath.Join(localDir, fileName))
+		if err != nil {
+			return fmt.Errorf("error creating file locally: %v", err)
+		}
+		// TODO: Check if there is some performance implication of multiple Open calls.
+		srcFile, err := sftp.Open(remoteDir + "\\" + fileName)
+
+		if err != nil {
+			// TODO: I am returning in case of error now when we fail for a single log file, let's see if it's ok to
+			// tolerate some errors here
+			return fmt.Errorf("error while opening remote directory on the Windows VM: %v", err)
+		}
+		_, err = io.Copy(dstFile, srcFile)
+		if err != nil {
+			return fmt.Errorf("error retrieving file %v from Windows VM: %v", fileName, err)
+		}
+		// flush memory
+		if err = dstFile.Sync(); err != nil {
+			return fmt.Errorf("error flusing memory: %v", err)
+		}
+		if err := srcFile.Close(); err != nil {
+			return fmt.Errorf("error closing file on the remote host %s", fileName)
+		}
+		if err := dstFile.Close(); err != nil {
+			return fmt.Errorf("error closing file %s locally", fileName)
```

Comment:
Should we error on these scenarios? I suggest we try to collect as many logs as we can and just log the failures. More over there is no reason to not close `srcFile` just because there was an error in closing `dstFile`.